edit-and-sync-features-with-feature-service/src/main/java/com/esri/arcgismaps/sample/editandsyncfeatureswithfeatureservice/components/MapViewModel.kt

```diff
@@ -0,0 +1,465 @@
+                        }.apply {
+                            // Set return attachments option to false
+                            // Indicates if any attachments are added to the geodatabase from the feature service
+                            returnAttachments = false
+                        }
+
+                // Create a generateGeodatabase job
+                generateGeodatabaseJob = geodatabaseSyncTask.createGenerateGeodatabaseJob(
+                    parameters = defaultParameters,
+                    pathToGeodatabaseFile = geodatabaseFilePath
+                )
+
+                generateGeodatabaseJob?.let { generateGeodatabaseJob ->
+
+                    // Show the dialog of generateGeodatabase Job
+                    showGenerateGeodatabaseJobProgressDialog = true
+
+                    // Create a flow-collection for the job's progress
+                    launch(Dispatchers.Main) {
+                        generateGeodatabaseJob.progress.collect { progress ->
+                            // Display the current job's progress value
+                            generateGeodatabaseJobProgress = progress
+                        }
+                    }
+
+                    // Start the job
+                    generateGeodatabaseJob.start()
+
+                    // If the job completed successfully, get the geodatabase from the result
+                    geodatabase = generateGeodatabaseJob.result().getOrElse {
+                        // Show an error and return if job failed
+                        messageDialogVM.showMessageDialog("Error fetching geodatabase: ${it.message}")
+                        // Dismiss the dialog
+                        showGenerateGeodatabaseJobProgressDialog = false
+                        // Clear any drawn boundary
+                        graphicsOverlay.graphics.clear()
+
+                        return@launch
+                    }
+
+                    // Load and display the geodatabase
+                    loadGeodatabase()
+
+                    // Dismiss the dialog
+                    showGenerateGeodatabaseJobProgressDialog = false
+                }
+            }
+        }
+    }
+
+    /**
+     * Loads the [Geodatabase] and renders the feature layers on to the [ArcGISMap]
+     */
+    private suspend fun loadGeodatabase() {
+        // Load the geodatabase
+        geodatabase?.let { geodatabase ->
+            geodatabase.load().onFailure {
+                // If the load failed, show the error and return
+                return messageDialogVM.showMessageDialog("Error loading geodatabase")
+            }
+
+            // Add all of the geodatabase feature tables to the map as feature layers
+            map.operationalLayers += geodatabase.featureTables.map { featureTable ->
+                FeatureLayer.createWithFeatureTable(featureTable)
+            }
+        }
```

Comment:
Nothing wrong with using an addition-assignment operator here, but I would suggest using `.addAll` operation from Kotlin Collections instead to help with the code readability: 

```suggestion
            map.operationalLayers.addAll(
                geodatabase.featureTables.map { featureTable ->
                    FeatureLayer.createWithFeatureTable(featureTable)
                }
            )
```